src/plot_api/plot_api.js

```diff
@@ -2429,6 +2429,9 @@ Plotly.relayout = function relayout(gd, astr, val) {
             var subplotIds;
             manageModeBar(gd);
 
+            Plotly.Fx.supplyLayoutDefaults(gd.layout, fullLayout, gd._fullData);
```

Comment:
`domodebar` (i.e. when only the `dragmode` or the `hovermode` changes) by-passes a **bunch** of things. So instead of running the whole `Plots.supplyDefaults` it only need to call the `Fx` defaults step and re-initialise the drag boxes.